lib/block-supports/duotone.php

```diff
@@ -398,8 +401,40 @@ function () use ( $svg ) {
 			echo $svg;
 		}
 	);
+}
 
-	return "url('#" . $filter_id . "')";
+/**
+ * Renders the duotone filter SVG for the preset and returns the filter url.
+ *
+ * @param array $preset Duotone preset value as seen in theme.json.
+ */
+function gutenberg_render_duotone_filter_preset( $preset ) {
```

Comment:
Kept this function so `value_func` in `PRESETS_METADATA` in [class-wp-theme-json-gutenberg.php](https://github.com/WordPress/gutenberg/pull/36236/files#diff-b03597cc3da199e5aa5a94950e8241135904853e7c3b82d758e42744878afae7) didn't have to change.

As a follow-up it may be able to be removed and replaced with `gutenberg_get_duotone_filter_property` when we can ensure that the SVG filters will be generated elsewhere.

Reply:
Hey, I'm still reviewing the changes to the block supports, so I don't fully understand your comment. However, if we plan/want to stop using `value_func` or the callback it's best to do it now. This is something that wasn't part of 5.8 ([see](https://github.com/WordPress/wordpress-develop/blob/5.8/src/wp-includes/class-wp-theme-json.php#L91)) so we can still change it now without worrying about backward compatibility issues (after 5.9 can be harder).